### DISCUSSION TITLE
Definition preview on ctrl + mouseover

### DISCUSSION THREAD

**Discussion by Petryanin** [microsoft/pylance-release#3156] [Enhancement]
Type: <b>Bug</b>

# Behaviour
## Expected vs. Actual

**Expected**: function/variable *definition* shows in a pop up window

**Actual**: only function/variable *name* shows in a pop up window

## Steps to reproduce:

1. Paste the code below into your workspace
```py
def func(*args, **kwargs):
    var = 1
    return var


if __name__ == "__main__":
    func()
```
2. Hold ctrl button
3. Place cursor over a function call or returning variable

# Diagnostic data

-   Python version (& distribution if applicable, e.g. Anaconda): 3.6.8
-   Type of virtual environment used (e.g. conda, venv, virtualenv, etc.): Global
-   Value of the `python.languageServer` setting: Pylance

<details>

<summary>Output for <code>Python</code> in the <code>Output</code> panel 
</summary>

<p>

```
LSP Notebooks experiment is enabled
LSP Notebooks interactive window support is enabled
> conda info --json
Python interpreter path: /bin/python3
> /usr/share/anaconda/bin/conda info --json
> /bin/python ~/.vscode-server/extensions/ms-python.python-2022.12.0/pythonFiles/get_output_via_markers.py ~/.vscode-server/extensions/ms-python.python-2022.12.0/pythonFiles/interpreterInfo.py
> /bin/python3 ~/.vscode-server/extensions/ms-python.python-2022.12.0/pythonFiles/get_output_via_markers.py ~/.vscode-server/extensions/ms-python.python-2022.12.0/pythonFiles/interpreterInfo.py
> /bin/python3.4 ~/.vscode-server/extensions/ms-python.python-2022.12.0/pythonFiles/get_output_via_markers.py ~/.vscode-server/extensions/ms-python.python-2022.12.0/pythonFiles/interpreterInfo.py
> /usr/bin/python ~/.vscode-server/extensions/ms-python.python-2022.12.0/pythonFiles/get_output_via_markers.py ~/.vscode-server/extensions/ms-python.python-2022.12.0/pythonFiles/interpreterInfo.py
> /usr/bin/python3 ~/.vscode-server/extensions/ms-python.python-2022.12.0/pythonFiles/get_output_via_markers.py ~/.vscode-server/extensions/ms-python.python-2022.12.0/pythonFiles/interpreterInfo.py
> /usr/bin/python3.4 ~/.vscode-server/extensions/ms-python.python-2022.12.0/pythonFiles/get_output_via_markers.py ~/.vscode-server/extensions/ms-python.python-2022.12.0/pythonFiles/interpreterInfo.py
Starting Pylance language server.
> /bin/python3 ~/.vscode-server/extensions/ms-python.python-2022.12.0/pythonFiles/linter.py -m pylint ~/help/test.py
cwd: ~/help
##########Linting Output - pylint##########

[]
```

</p>
</details>

<details>

<summary>User Settings</summary>

<p>

```
Multiroot scenario, following user settings may not apply:

languageServer: "Pylance"

sortImports
• args: "<placeholder>"

formatting
• autopep8Args: "<placeholder>"

```

</p>
</details>

Extension version: 2022.12.0
VS Code version: Code 1.70.0 (da76f93349a72022ca4670c1b84860304616aaa2, 2022-08-04T04:38:16.462Z)
OS version: Windows_NT x64 10.0.19043
Modes:
Remote OS version: Linux x64 3.10.0-1127.18.2.el7.x86_64

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-6500 CPU @ 3.20GHz (4 x 3192)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.96GB (9.35GB free)|
|Process Argv|--crash-reporter-id cf06ce02-264e-42b2-a621-31424cdada19|
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|SSH: work|
|OS|Linux x64 3.10.0-1127.18.2.el7.x86_64|
|CPUs|Intel(R) Core(TM) i7-7700 CPU @ 3.60GHz (8 x 3999)|
|Memory (System)|31.13GB (1.58GB free)|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
pythonvspyl392:30443607
vserr242cf:30382550
pythontb:30283811
vsjup518:30340749
pythonvspyt551cf:30345471
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscoreces:30445986
pythondataviewer:30285071
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
vsc1dst:30438360
pythonvs932:30410667
wslgetstarted:30449410
vscscmwlcmt:30465135
cppdebug:30492333
pylanb8912:30529769
vsclangdf:30486550
c4g48928:30535728
hb751961:30541207
dsvsc012:30540252

```

</details>

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.

**Comment by heejaechang** [comment]
Pylance currently return Location, but it is asking returning LocationLink (https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#locationLink)
